internal/caller/linux/linuxconfig.go

```diff
@@ -0,0 +1,136 @@
+package caller
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+var (
+	installPath   = "/etc/wireguard"
+	tunnelTemplate = `[Interface]
+PrivateKey = %s
+Address = %s
+	
+[Peer]
+PublicKey = %s
+AllowedIPs %s
+Endpoint = %s`
+)
+
+// LinuxConfig represents the tunnel configuration for a linux machine.
+type LinuxConfig struct {
+	localDevice wgtypes.Device
+	tunnel      string
+	confName    string
+	allowedIps  string
+	addr        string
+	endpoint    string
+}
+
+// init transmits the linux config factory to the caller Client.
+func init() {
+	AppConfigurator = NewConfig()
+}
+
+// NewConfig generates and returns a new LinuxConfig.
+func NewConfig() *LinuxConfig {
+	newDevice := wgtypes.Device{
+		Name: "Linux Configurator",
+		Type: wgtypes.Userspace,
+	}
+
+	newLinuxConfig := LinuxConfig{
+		localDevice: newDevice,
+		tunnel:      tunnelTemplate,
+		confName:    "",
+	}
+	return &newLinuxConfig
+}
+
+// GenerateKeys creates a new set of Private and Public Keys for the
+// LinuxConfig.
+func (l *LinuxConfig) GenerateKeys() error {
```

Comment:
Since every driver will need to be able to generate keys, consider putting this and other similar functions into a base implementation of the Configurator which can be embedded in this one.

Reply:
I have a base configuration type in progress. I will make a pr for it as after the tunnel pr has finished code review and is accepted.